src/v/archival/tests/archiver_operations_impl_gtest.cc

```diff
@@ -0,0 +1,670 @@
+// Copyright 2024 Redpanda Data, Inc.
```

Comment:
nit: could you name this `archiver_operations_gtest_utils.cc` or something that indicates that there are no tests in this file and that it's just utils?

Reply:
this commit just adds some basic defs and the following commit will be adding tests